external-crates/move/crates/move-compiler/src/diagnostics/mod.rs

```diff
@@ -806,30 +808,49 @@ impl Migration {
         const NEEDS_PUBLIC: u8 = codes::Migration::NeedsPublic as u8;
         const NEEDS_BACKTICKS: u8 = codes::Migration::NeedsRestrictedIdentifier as u8;
         const NEEDS_GLOBAL_QUAL: u8 = codes::Migration::NeedsGlobalQualification as u8;
+        const REMOVE_FRIEND: u8 = codes::Migration::RemoveFriend as u8;
+        const MAKE_PUB_PACKAGE: u8 = codes::Migration::MakePubPackage as u8;
 
-        let (file_id, line, col) = self.find_file_location(&diag);
+        let (file_id, (start_line, start_col), (end_line, end_col)) =
+            self.find_file_location(&diag);
         let file_change_entry = self.changes.entry(file_id).or_default();
-        let line_change_entry = file_change_entry.entry(line).or_default();
+        let line_change_entry = file_change_entry.entry(start_line).or_default();
         match (diag.info().category(), diag.info().code()) {
-            (CAT, NEEDS_MUT) => line_change_entry.push((col, MigrationChange::AddMut)),
-            (CAT, NEEDS_PUBLIC) => line_change_entry.push((col, MigrationChange::AddPublic)),
+            (CAT, NEEDS_MUT) => line_change_entry.push((start_col, MigrationChange::AddMut)),
+            (CAT, NEEDS_PUBLIC) => line_change_entry.push((start_col, MigrationChange::AddPublic)),
             (CAT, NEEDS_BACKTICKS) => {
                 let old_name = diag.primary_msg().to_string();
-                line_change_entry.push((col, MigrationChange::Backquote(old_name)))
+                line_change_entry.push((start_col, MigrationChange::Backquote(old_name)))
             }
             (CAT, NEEDS_GLOBAL_QUAL) => {
-                line_change_entry.push((col, MigrationChange::AddGlobalQual))
+                line_change_entry.push((start_col, MigrationChange::AddGlobalQual))
+            }
+            (CAT, REMOVE_FRIEND) => {
+                if start_line == end_line {
+                    line_change_entry.push((end_col, MigrationChange::RemoveFriend(start_col)))
+                }
+            }
+            (CAT, MAKE_PUB_PACKAGE) => {
+                if start_line == end_line {
+                    line_change_entry.push((end_col, MigrationChange::MakePubPackage(start_col)))
```

Comment:
For consistency, would it be possible to still push `start_col` ad the key and place `end_col` in the variants?

Reply:
Should we do something else if we can't do this migration? Perhaps emit an error or similar?

Reply:
> For consistency, would it be possible to still push `start_col` ad the key and place `end_col` in the variants?

I tried, but it is difficult because we reverse iterate over the string

Reply:
> Should we do something else if we can't do this migration? Perhaps emit an error or similar?

I didn't see a spot to do so. Should I just panic? Insert a comment into the output? Something else?